### ISSUE TITLE
Host Interop does not pass null arguments to `Function`

### ISSUE THREAD

**Issue by kustosz** [oracle/graal#4741]
**Describe GraalVM and your environment :**
 - GraalVM version or commit id if built from source: **[e.g. 19.3]**
 - CE or EE: **CE**
 - JDK version: **JDK11**
 - OS and OS Version: **macOS Monterey**
 - Architecture: **arm64**
 - The output of `java -Xinternalversion`:
```
OpenJDK 64-Bit Server VM (11.0.13+7-jvmci-21.3-b05) for bsd-amd64 JRE (11.0.13+7-jvmci-21.3-b05), built on Oct 15 2021 23:02:45 by "graal1" with clang 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)
```

**Have you verified this issue still happens when using the latest snapshot?**
You can find snapshot builds here: https://github.com/graalvm/graalvm-ce-dev-builds/releases
**No**

**Describe the issue**
When passing a function to java through interop, `function.apply(null)` results in the function being called with no arguments, rather than a single `null` argument. This breaks languages for which function arity is meaningful. 

**Code snippet or code repository that reproduces the issue**
Test.java:
```java
import java.util.function.Function;

public class Test {
  public void call(Function<Object, Object> fn) {
    fn.apply(null);
    fn.apply(0);
  }
}
```
test.js:
```js
Java.addToClasspath(".");
Test = Java.type("Test");
new Test().call(function () { console.log(JSON.stringify(arguments)); });
```

**Steps to reproduce the issue**
1. Run `javac Test.java`
2. Run `js --polyglot --jvm test.js`
3. It prints:
```
{}
{"0":0}
```

**Expected behavior**
The first call should be passed a single `null` argument. Note that this repro uses JS for convenience (as the easiest-to-get truffle language). While this issue is not at all serious for JS, for languages that validate function arities this is very unexpected (and impossible to work around).

**Additional context**
Add any other context about the problem here. Specially important are stack traces or log output. Feel free to link to gists or to screenshots if necesary

**Comment by woess** [comment]
Thanks for the report! This issue should be fixed by https://github.com/oracle/graal/commit/f8a2bd7d79cdd7c7afbe6851c21b2876a0946a36.

**Comment by JaroslavTulach** [comment]
Thanks a lot Andreas for a quick fix! As GraalVM EE 22.2 seems to be already out, I assume the fix will get into GraalVM CE/EE 22.3 in middle of October.

Is there any chance to get it sooner? Any change to get it into 21.3.x update? Enso is still running on 21.3... CCing @ezzarghili

**Comment by woess** [comment]
It should be possible to backport this to 21.3.x.
